### ISSUE TITLE
glGetActiveUniformBlockiv not functioning for vector output

### ISSUE THREAD

**Issue by FreakingBarbarians** [libgdx/libgdx#6203]
Related PR https://github.com/libgdx/libgdx/pull/6208

#### Issue details
When querying vector information from UniformBlocks (i.e. GL_UNIFORM_BLOCK_ACTIVE_UNIFORM_INDICES) only the first item is ever written.

#### Reproduction steps/code
I don't have time to create a reproduction rn, but my stacktrace looks something like this
```
My code:

                val indices = allocateIntBuffer(count) // direct byte buffer as an int buffer
                indices.rewind()
                gl.glGetActiveUniformBlockiv( // gl is typealiased to Gdx.gl30
                    programId,
                    i, // index of the ubo
                    GL30.GL_UNIFORM_BLOCK_ACTIVE_UNIFORM_INDICES,
                    indices
                )
Libgdx: Lwjgl3GL30.java:484

	@Override
	public void glGetActiveUniformBlockiv (int program, int uniformBlockIndex, int pname, IntBuffer params) {
		params.put(GL31.glGetActiveUniformBlocki(program, uniformBlockIndex, pname));
	}
Lwjgl: GL31C.java:584

    @NativeType("void")
    public static int glGetActiveUniformBlocki(@NativeType("GLuint") int program, @NativeType("GLuint") int uniformBlockIndex, @NativeType("GLenum") int pname) {
        MemoryStack stack = stackGet(); int stackPointer = stack.getPointer();
        try {
            IntBuffer params = stack.callocInt(1);
            nglGetActiveUniformBlockiv(program, uniformBlockIndex, pname, memAddress(params));
            return params.get(0);
        } finally {
            stack.setPointer(stackPointer);
        }
    }
```
I may provide a proper reproduction later on.

#### Version of LibGDX and/or relevant dependencies
1.9.11
#### Stacktrace
```java
glGetActiveUniformBlocki:553, GL31 (org.lwjgl.opengl)
glGetActiveUniformBlockiv:485, Lwjgl3GL30 (com.badlogic.gdx.backends.lwjgl3)
introspect:140, ShaderProgram (com.nanite.graphics)
. . .
initializeListener:433, Lwjgl3Window (com.badlogic.gdx.backends.lwjgl3)
update:381, Lwjgl3Window (com.badlogic.gdx.backends.lwjgl3)
loop:138, Lwjgl3Application (com.badlogic.gdx.backends.lwjgl3)
<init>:114, Lwjgl3Application (com.badlogic.gdx.backends.lwjgl3)
main:18, DesktopLauncher (com.cellular.game.desktop)
```

#### Please select the affected platforms
- [ ] Android
- [ ] iOS
- [ ] HTML/GWT
- [x] Windows
- [ ] Linux
- [ ] MacOS
(although I imagine other platforms will be affected as well)

I imagine it's a simple matter of calling the wrong jwjgl function.

**Comment by tommyettinger** [comment]
Are you maybe using a different ShaderProgram class? The stacktrace has `introspect:140, ShaderProgram (com.nanite.graphics)`, and libGDX has its own ShaderProgram that is not in a com.nanite.graphics package. This result is still not  what I'd expect; the pasted libGDX code only puts one `int` into the `params`, but there could be another method that outputs multiple values.

**Comment by FreakingBarbarians** [comment]
Yeah, sorry I should have specified. I'm currently building some introspection on top of the ShaderProgram class in libgdx.
```kotlin
...
import com.badlogic.gdx.graphics.glutils.ShaderProgram as GdxShaderProgram
...
class ShaderProgram(
    val vertexShader: String,
    val fragmentShader: String
) : Disposable {
...
val gdx by lazy {
        GdxShaderProgram(vertexShader, fragmentShader).apply {
            if (!isCompiled()) {
                throw IllegalArgumentException("""Error compiling shader:
--------- vertex shader ----------
${vertexShader}
--------- fragment shader ----------
${fragmentShader}
--------- error ----------
${getLog()}
""")
            }
        }
    }
...
}
```

It wraps around the ShaderProgram inside of Libgdx.
But apart from that I would not expect this to affect the return value of `glGetActiveUniformBlockiv`

**Comment by FreakingBarbarians** [comment]
The problem as I see it is that, the vectorized `glGetActiveUniformBlockiv` calls the unvectorized version `glGetActiveUniformBlocki`
Although in the opengl spec I only see `glGetActiveUniformBlockiv`, in the LWJGL api both exist.

**Comment by FreakingBarbarians** [comment]
Update:
After wrangling around with gradle for a very long time I managed to get a build of libgdx from source working with my project.

Changing the function call like this fixed the problem for me.
```java
Libgdx: Lwjgl3GL30.java:484
	@Override
	public void glGetActiveUniformBlockiv (int program, int uniformBlockIndex, int pname, IntBuffer params) {
		params.put(GL31.glGetActiveUniformBlocki(program, uniformBlockIndex, pname));
	}
->
	@Override
	public void glGetActiveUniformBlockiv (int program, int uniformBlockIndex, int pname, IntBuffer params) {
		GL31.glGetActiveUniformBlockiv(program, uniformBlockIndex, pname, params);
	}
```

**Comment by mgsx-dev** [comment]
indeed, it looks like a copy/paste from Lwjgl2 backend which was also wrong (or not available at this time).
- it seams like Lwjgl2 backend should call GL31.glGetActiveUniformBlock (without i) passing IntBuffer as parameter.
- and Lwjgl3 backend should call GL31.glGetActiveUniformBlockiv as you mentionned.

@FreakingBarbarians could you make the PR? :-)

**Comment by FreakingBarbarians** [comment]
Sure I could have something ready by tomorrow-ish

**Comment by mgsx-dev** [comment]
fixed by #6208
